syntax = "proto3";

import "google/protobuf/wrappers.proto";

package ecommerce.v1;

option go_package = "github.com/omihirofumi/grpc-up-and-running/ecommerce_v1";

service OrderManagement {
  rpc searchOrders(google.protobuf.StringValue) returns (stream Order);
  rpc addOrder(Order) returns (google.protobuf.StringValue);
  rpc getOrder(google.protobuf.StringValue) returns (Order);
  rpc updateOrders(stream Order) returns (google.protobuf.StringValue);
  rpc processOrders(stream google.protobuf.StringValue) returns (stream CombinedShipment);
}

message Order {
  string id = 1;
  repeated string items = 2;
  float price = 4;
  string destination = 5;
}

message CombinedShipment {
  string id = 1;
  string status = 2;
  repeated Order ordersList = 3;
}